cpp/elasticity.cpp

```diff
@@ -34,4 +34,93 @@ void dolfinx_contact::compute_normal_strain_basis(
       }
     }
   }
+}
+
+//-----------------------------------------------------------------------------
+void dolfinx_contact::compute_sigma_n_basis(xt::xtensor<double, 3>& sig_n,
```

Comment:
I guess all of these functions could be templated for gdim/bs. Would improve the code quite alot performance wise